server/middlewares/status.js

```diff
@@ -0,0 +1,41 @@
+// Check and set if festival is free for unregistered Visitors
+
+import httpStatus from 'http-status'
+import passport from '../services/passport'
+
+import { APIError } from '../helpers/errors'
+import { getConfig } from '../config'
+
+function checkStatus(checkUser, checkTicketPrice, req, res, next) {
+  if (checkUser && !req.user) {
+    passport.authenticate('jwt', { session: false }, (err, user) => {
+      if (err) {
+        return next(err)
+      }
+      if (!user) {
+        req.user = false
+      }
+      req.user = user
+      return next()
+    })(req, res, next)
+  }
+
+  if (checkTicketPrice) {
+    getConfig('festivalTicketPrice').then(config => {
+      if (config.festivalTicketPrice === 0) {
+        req.freeFestival = true
```

Comment:
... and this one throwing an error when the user is not a visitor / participant when the festival ticket price is > 0